src/vue/forms/ChangePasswordForm.vue

```diff
@@ -0,0 +1,214 @@
+          :label="'change-password-form.new-password-lbl' | globalize"
+          :disabled="formMixin.isDisabled"
+        />
+      </div>
+    </div>
+    <div class="app__form-row">
+      <div class="app__form-field">
+        <input-field
+          v-model="form.confirmPassword"
+          @blur="touchField('form.confirmPassword')"
+          id="change-password-confirm"
+          type="password"
+          :error-message="getFieldErrorMessage('form.confirmPassword')"
+          :label="'change-password-form.confirm-password-lbl' | globalize"
+          :disabled="formMixin.isDisabled"
+        />
+      </div>
+    </div>
+    <div
+      v-if="isTotpEnabled"
+      class="app__form-row"
+    >
+      <div class="app__form-field">
+        <input-field
+          v-model="form.tfaCode"
+          @blur="touchField('form.tfaCode')"
+          id="change-password-tfa-code"
+          :error-message="getFieldErrorMessage('form.tfaCode')"
+          :label="'change-password-form.tfa-code-lbl' | globalize"
+          :disabled="formMixin.isDisabled"
+        />
+      </div>
+    </div>
+
+    <div class="app__form-actions">
+      <form-confirmation
+        v-if="formMixin.isConfirmationShown"
+        @ok="hideConfirmation() || submit()"
+        @cancel="hideConfirmation"
+      />
+      <button
+        v-else
+        v-ripple
+        type="submit"
+        class="change-password-form__submit-btn"
+        :disabled="formMixin.isDisabled"
+      >
+        {{ 'change-password-form.change-password-btn' | globalize }}
+      </button>
+    </div>
+  </form>
+</template>
+
+<script>
+import FormMixin from '@/vue/mixins/form.mixin'
+
+import { required, requiredIf, password, sameAs } from '@validators'
+
+import { ErrorHandler } from '@/js/helpers/error-handler'
+import { Bus } from '@/js/helpers/event-bus'
+
+import { Sdk } from '@/sdk'
+import { errors } from '@tokend/js-sdk'
+
+import { vuexTypes } from '@/vuex'
+import { mapGetters, mapActions } from 'vuex'
+
+export default {
+  name: 'change-password-form',
+  mixins: [FormMixin],
+  data: _ => ({
+    form: {
+      currentPassword: '',
+      newPassword: '',
+      confirmPassword: '',
+      tfaCode: '',
+    },
+  }),
+  validations: {
+    form: {
+      currentPassword: { required, password },
+      newPassword: { required, password },
+      confirmPassword: {
+        required,
+        password,
+        sameAsPassword: sameAs(function () { return this.form.newPassword }),
+      },
+      tfaCode: {
+        required: requiredIf(function () { return this.isTotpEnabled }),
+      },
+    },
+  },
+  computed: {
+    ...mapGetters({
+      wallet: vuexTypes.wallet,
+      isTotpEnabled: vuexTypes.isTotpEnabled,
+    }),
+  },
+  methods: {
+    ...mapActions({
+      storeWallet: vuexTypes.STORE_WALLET,
+    }),
+    async submit () {
+      if (!this.isFormValid()) {
+        return
+      }
+      this.disableForm()
+      try {
```

Comment:
@quinn71, here goes absolute try-catch hell. not sure we got time to refactor it now, so I created a task for it:
https://app.asana.com/0/894507180698157/1108030165482984/f

Yeah refactor tasks after refactor...